docs/inference_api_beta/python/gretel.py

```diff
@@ -0,0 +1,181 @@
+from __future__ import annotations
+
+import json
+import time
+import uuid
+from dataclasses import dataclass
+from typing import Any, Iterator, Optional
+
+from gretel_client import configure_session
+from gretel_client.config import GretelClientConfigurationError, get_session_config
+from gretel_client.users.users import get_me
+
+
+class GretelValidationError(Exception):
+    pass
+
+
+def validate_gretel() -> Optional[str]:
+    try:
+        configure_session(validate=True)
+    except GretelClientConfigurationError:
+        raise GretelValidationError(
+            "Could not authenticate to Gretel. Please verify your API key is configured."
+        )
+
+    user_dict = get_me()
+    return user_dict["email"]
+
+
+MAX_SIZE = 50
+
+
+@dataclass
+class TabLLMRequest:
+    model_id: str
+    prompt: str
+    temperature: float
+    top_k: int
+    top_p: float
+
+
+class TabLLMStreamError(Exception):
+    pass
+
+
+class TabLLMStream:
+    request: TabLLMRequest
+    target_count: int
+    generated_count: int = 0
+    stream_meta_id: str
+
+    _curr_stream_id: Optional[str] = None
+    _next_iter: Optional[str] = None
+
+    def __init__(self, *, request: TabLLMRequest, record_count: int):
+        self.request = request
+        self.target_count = record_count
+        self.stream_meta_id = uuid.uuid4().hex
+
+    def _reset_stream(self) -> None:
+        self._curr_stream_id = None
+        self._next_iter = None
+
+    def _create_stream(self) -> None:
+        """
+        Create a new TabLLM stream via the Gretel API
+        """
+        payload = {
+            "model_id": self.request.model_id,
+            "num_rows": min(MAX_SIZE, self.target_count - self.generated_count),
+            "params": {
+                "temperature": self.request.temperature,
+                "top_k": self.request.top_k,
+                "top_p": self.request.top_p,
+            },
+            "prompt": self.request.prompt,
+        }
+
+        print("Creating Gretel TabLLM inference stream.")
+        resp = _do_api_call("post", "/v1/inference/tabular/stream", body=payload)
+        self._curr_stream_id = resp.get("stream_id")
+        self._next_iter = None
+        print(f"Created Gretel TabLLM inference stream: {self._curr_stream_id}")
+
+    def iter_stream(self) -> Iterator[dict[str, Any]]:
+        while self.generated_count < self.target_count:
+            # Make an API request for a new Gretel stream if need be
```

Comment:
How are we preventing duplicates after 50 rows if we're starting a new stream?

Reply:
We don't